### ISSUE TITLE
ddcutil: Install udev rool to allow group-based monitor brightness setting

### ISSUE THREAD

**Issue by nh2** [NixOS/nixpkgs#91771]
[`ddcutil`](https://www.ddcutil.com/) and its GUI `ddcui` allow adjusting the hardware settings (e.g. brightness) of external monitors.

This works with `root` everywhere, but a better way (on NixOS) is to install the upstream udev rule ([see here](https://github.com/rockowitz/ddcui/issues/20#issuecomment-650829006)) so that users of an `i2c` group can adjust monitor brighness without having to be root.

It would be cool for somebody to try that udev rule, and up in NixOS accordingly.

**Comment by rnhmjoj** [comment]
We could create a `hardware.ddcutil.enable` option to install the rule but It's fairly trivial to do manually. I don't have an ad-hoc group for i2c but I'm using wheel, in any case it's just
```
services.udev.extraRules = ''
  # allow admin use of i2c devices
  ACTION=="add", KERNEL=="i2c-[0-9]*", GROUP="wheel", MODE="666"
'';
```

**Comment by rnhmjoj** [comment]
Still relevant, I think. So, @nh2 how do you propose the udev rule should be installed?

**Comment by colemickens** [comment]
It seems other distros have an `i2c` group and then install a udev rule granting permission to that group. That seems fairly straightforward and implementable in nixos.

I think this would also enable folks to use `gddccontrol`/`ddccontrol` without root, by adding themselves to the `i2c` that would then exist. The group/rule could be behind `hardware.i2c.enable`, which could also add `i2c-dev` to system modules so that dependency would also be taken care of.

cc: @lovesegfault

**Comment by rnhmjoj** [comment]
Fixed by #111996.
It should be enough to do `hardware.i2c.enable = true` to use ddcutil/ddcui on a desktop computer.
